### PR TITLE
MX-13962: consistent tokens values length checks

### PR DESCRIPTION
**Pull request by bogdan-rosianu**
In the case of large tokens values, some error messages were wrong. This PR brings consistency by using a flag that ensures the backward compatibility

### CODE REVIEW

**Review by sasurobert — DISMISSED** [review]

File: builtInFunctions/esdtLocalBurn_test.go [hunk 1]
```diff
@@ -118,6 +126,63 @@ func TestEsdtLocalBurn_ProcessBuiltinFunction_CannotAddToEsdtBalanceShouldErr(t
...
+			return &mock.DataTrieTrackerStub{
+				RetrieveValueCalled: func(_ []byte) ([]byte, uint32, error) {
+					esdtData := &esdt.ESDigitalToken{Value: big.NewInt(100)}
+					serializedEsdtData, err := marshaller.Marshal(esdtData)
+					return serializedEsdtData, 0, err
+				},
+				SaveKeyValueCalled: func(key []byte, value []byte) error {
+					esdtData := &esdt.ESDigitalToken{}
+					_ = marshaller.Unmarshal(esdtData, value)
+					//require.Equal(t, big.NewInt(101), esdtData.Value)
```
Comment (iulianpascalau) [inline builtInFunctions/esdtLocalBurn_test.go:152]: commented code & benign code in this function

File: builtInFunctions/esdtLocalBurn.go [hunk 2]
```diff
@@ -79,6 +86,11 @@ func (e *esdtLocalBurn) ProcessBuiltinFunction(
 		return nil, err
 	}
 
+	if e.enableEpochsHandler.IsConsistentTokensValuesLengthCheckEnabled() {
+		if len(vmInput.Arguments[1]) > core.MaxLenForESDTIssueMint {
```
Comment (iulianpascalau) [inline builtInFunctions/esdtLocalBurn.go:90]: can we rename the core.MaxLenForESDTIssueMint const to something else as now it is used in alot of other functions. Something like: core.MaxLenForESDTValue ?

File: builtInFunctions/esdtLocalBurn_test.go (hunk 1 above)
Comment (bogdan-rosianu) [reply builtInFunctions/esdtLocalBurn_test.go:152]: removed

File: builtInFunctions/esdtLocalBurn.go (hunk 2 above)
Comment (bogdan-rosianu) [reply builtInFunctions/esdtLocalBurn.go:91]: added a TODO for this

**Review by iulianpascalau — APPROVED** [review]

**Review by sasurobert — APPROVED** [review]

**Pull request merged**